lib/ansible/plugins/cliconf/ios.py

```diff
@@ -210,6 +210,10 @@ def get_device_info(self):
         match = re.search(r'^Cisco (.+) \(revision', data, re.M)
         if match:
             device_info['network_os_model'] = match.group(1)
+        else:
+            match = re.search(r'^[Cc]isco (\S+).+bytes of .*memory', data, re.M)
+            if match:
+                device_info['network_os_model'] = match.group(1)
 
```

Comment:
This can be simplified as:
```
search_strs = ['^Cisco (.+) \(revision', '^[Cc]isco (\S+).+bytes of .*memory']
for item in search_strs:
   match = re.search(r'%s' % item, data, re.M)
   if match:
       device_info['network_os_model'] = match.group(1)
       break
```